### DISCUSSION TITLE
Running Rosetta in non-ubuntu distributions (Fedora, OpenSUSE Tumbleweed)

### DISCUSSION THREAD

**Discussion by willaaam** [utmapp/UTM#5233] [Show and tell]
Just wanted to drop a line, took me a while to figure this out. Not all distributions come with the tools (bimfmt-update) in the UTM documentation to run rosetta in Linux. This works for Fedora and OpenSUSE:

Register handler for x64 binaries (needs to be run as root and run on every reboot -- you can write a service to run this every boot):
echo ":rosetta:M:0:\x7fELF\x02\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\x3e\x00:\xff\xff\xff\xff\xff\xfe\xfe\x00\xff\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff\xff:/media/rosetta/rosetta:OCF" > /proc/sys/fs/binfmt_misc/register

Deregister:
echo -1 >  /proc/sys/fs/binfmt_misc/rosetta

**Comment by hasan4791** [comment]
@willaaam Not able to get this working on Fedora 38 Cloud image. I can register the rosetta with binfmt but podman is shutting down the engine without any error!!.

**Reply by Linus-XZX** [reply]
Was in a similar spot recently and I believe the issue is SELinux.

The virtiofs image for Rosetta is readonly (or `Operation not permitted` if remounted as rw and attempting to write), which results in the executables not being assigned security labels (`unlabeled_t`). Accessing them from within containers (`container_t`) therefore presumably constitutes some form of security boundary cross.
